symex-interface-arc.el

```diff
@@ -80,6 +81,20 @@ Accounts for different point location in evil vs Emacs mode."
   "Evaluate buffer."
   (error "Not implemented"))
 
+(defvar symex-arc-modes 'arc-mode)
+
+(symex-interface-extend
+ symex-arc-modes
+ (list
+   :eval #'symex-eval-arc
+   :eval-definition #'symex-eval-definition-arc
+   :eval-pretty #'symex-eval-pretty-arc
+   :eval-thunk #'symex-eval-thunk-arc
+   :eval-print #'symex-eval-print-arc
+   :describe-symbol #'symex-describe-symbol-arc
+   :repl #'symex-repl-arc
+   :run #'symex-run-arc))
```

Comment:
Should be wrap those forms into thunks ?
That may be called in the `symex-interface-builtins.el` (which would justify it a little more)

Reply:
Do you see any benefits from wrapping these in thunks? Otherwise, it seems fine as is.

Reply:
The thunk wrapping could bring some flexibility maybe ?
Maybe there is some subtelties that I don't get though. 
Any concrete thoughts you have on this ?

Reply:
Do you mean something like this:

```
:eval (lambda () #'symex-eval-arc)
```

I can't think of any reason to do this, so let's leave it as is. If we find we need more flexibility in the future we can always add what we need then. "YAGNI," as they say.

Reply:
I am sorry I confused my self, I took [your answer](https://github.com/drym-org/symex.el/pull/136#discussion_r1632184240) for a comment, forgetting about my initial question...
Ok :) so let's keep it like it is. 
I was talking about wrapping the whole `symex-interface-extend` form into a thunk.

```
(defun start-symex-arc ()
  (symex-interface-extend
   symex-arc-modes
   (list
    :eval #'symex-eval-arc
    :eval-definition #'symex-eval-definition-arc
    :eval-pretty #'symex-eval-pretty-arc
    :eval-thunk #'symex-eval-thunk-arc
    :eval-print #'symex-eval-print-arc
    :describe-symbol #'symex-describe-symbol-arc
    :repl #'symex-repl-arc
    :run #'symex-run-arc)))
```

Reply:
Ah I see, that's a good point. I am also uneasy about having things happen at module loading time instead of via an explicit function invocation at the appropriate time. Are you thinking something like this:

`symex-interface-arc.el`:
```
(defun symex-interface-register-arc ()
  "Register the Arc runtime interface."
  (symex-interface-extend
   symex-arc-modes
   (list
    :eval #'arc-send-last-sexp
    :eval-definition #'arc-send-definition
    :eval-pretty #'arc-send-last-sexp
    :eval-thunk #'symex-eval-thunk-arc
    :repl #'symex-repl-arc
    :switch-to-scratch-buffer #'symex-switch-to-scratch-buffer-arc)))
```

`symex-interface-builtins.el`:
```
(defun symex-register-builtin-interfaces ()
  "Register built-in runtime interfaces."
  (symex-interface-register-arc)
  ...)
```

And then `symex-register-builtin-interfaces` could be called in `symex-initialize`.

Yes, I think something along these lines would be a good idea.